pkg/ring/batch.go

```diff
@@ -70,6 +72,10 @@ func DoBatch(ctx context.Context, op Operation, r ReadRing, keys []uint32, callb
 		cleanup()
 		return fmt.Errorf("DoBatch: InstancesCount <= 0")
 	}
+
+	span, _ := opentracing.StartSpanFromContext(ctx, "DoBatch")
+	defer span.Finish()
```

Comment:
Umm, do we need this span? This method will be called by other components.
Is this a time consuming operation?

Reply:
The idea on the DoBatch span is to check the time we spend preparing ingester requests (itemTrackers). In some load test, I saw that preparing requests can take some time. This could help get more visibility on when and why this happens.
About using in other points, distributor(quorum) and alertmanager also use the doBatch. I did check cpu and latency after adding new samples but didnt notice any concerns. Another easy approach is to wrap the doBatch call on distributor in an inner func and add a trace only in that scenario. wdyt?

Reply:
> Another easy approach is to wrap the doBatch call on distributor in an inner func and add a trace only in that scenario. wdyt?

Yeah I prefer this approach.